### ISSUE TITLE
Operator '[]' is not applicable to an Undef Value.

### ISSUE THREAD

**Issue by Pwd9000-ML** [keirans/azuremetadata#23]
Having some difficulty with this module with Puppet Bolt.
I have added the mod to my puppet bolt env and puppetfile.
When I run my bolt plan to obtain the facter fact of the target:
E.g:
$facts['az_metadata']

It seems that the remote target cannot parse the module.
Error:
C:/Program Files/Puppet Labs/Bolt/lib/ruby/gems/2.5.0/gems/puppet-6.4.0/lib/puppet/pops/evaluator/runtime3_support.rb:39:in `optionally_fail': Evaluation Error: Operator '[]' is not applicable to an Undef Value. 

The module works without issue in my PE environment.
However I would like to know the configuration or integration steps of this module into Puppet Bolt.

**Comment by Pwd9000-ML** [comment]
Worked with the puppet team to get this working in a bolt environment.
have a suitable solution now and is no longer relevant.
Module works as expected in a bolt environment.

**Comment by keirans** [comment]
Thanks for the heads up on this.

I am not using bolt at the moment, however if you could share any information about things to be aware if it would be great.

**Comment by Pwd9000-ML** [comment]
No worries.
It all came down to module dependencies. As bolt uses its own puppetfile.
It seems that this module was incompatible with another module (or dependent) on another module.
The module:
`mod 'puppetlabs-puppet_agent', '1.6.2'` which was what I was using in Bolt did not have the relevant tasks for bolt to consume this module via the `apply_prep` function on my target nodes for: `mod 'keirans-azuremetadata', '0.1.9'` 
So I had to bump my module version up to `mod 'puppetlabs-puppet_agent', '2.1.2'`

Since bolt uses a function called `apply_prep` to consume the modules of bolt I had to bump up the puppet_agent module for bolt to successfully utilize function `$nodes.apply_prep` which then consumed the az metadata module correctly.
